src/main/java/zd/zero/waifu/motivator/plugin/listeners/ExitCodeListener.kt

```diff
@@ -0,0 +1,71 @@
+package zd.zero.waifu.motivator.plugin.listeners
+
+import com.intellij.execution.ExecutionListener
+import com.intellij.execution.ExecutionManager
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import zd.zero.waifu.motivator.plugin.ProjectConstants
+import zd.zero.waifu.motivator.plugin.alert.AlertConfiguration
+import zd.zero.waifu.motivator.plugin.assets.VisualMotivationAssetProvider
+import zd.zero.waifu.motivator.plugin.assets.WaifuAssetCategory
+import zd.zero.waifu.motivator.plugin.motivation.VisualMotivationFactory
+import zd.zero.waifu.motivator.plugin.onboarding.UpdateNotification.sendMessage
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorPluginState
+import zd.zero.waifu.motivator.plugin.tools.doOrElse
+
+class ExitCodeListener(private val project: Project) : Runnable, Disposable {
+    private val messageBus = ApplicationManager.getApplication().messageBus.connect()
+
+    init {
+        messageBus.subscribe(ExecutionManager.EXECUTION_TOPIC, object : ExecutionListener {
+            override fun processTerminated(
+                executorId: String,
+                env: ExecutionEnvironment,
+                handler: ProcessHandler,
+                exitCode: Int
+            ) {
+                if (exitCode > 0 && env.project == project) {
```

Comment:
I think we can change this to `exitCode != 0` as there are processes that return `-1` for failure event.

Reply:
Yeah I initially started that route, however, I think failed test exit with at exit code of -1. Which means more waifu notifications. I've got somethings I need to investigate though.

Reply:
Okay, so I changed it to `exitCode != 0`

Things to keep in mind:
- When a test fails to run it:
    - Emits a non-zero exit code motivation event
    - emits a test failure motivation event

So Wendi will just wait until events stop coming in then react to the last event received.